src/python.cc

```diff
@@ -2121,6 +2328,396 @@ ModelState::ModelState(TRITONBACKEND_Model* triton_model)
+
+    std::unique_ptr<PbString> pb_string =
+        PbString::Create(shm_pool_, model_config);
+    bi::managed_external_buffer::handle_t string_handle =
+        pb_string->ShmHandle();
+
+    auto_complete_message->Args() = string_handle;
+    stub_message_queue_->Push(auto_complete_message->ShmHandle());
+
+    std::unique_ptr<IPCMessage> auto_complete_response_message =
+        IPCMessage::LoadFromSharedMemory(
+            shm_pool_, parent_message_queue_->Pop());
+
+    if (auto_complete_response_message->Command() !=
+        PYTHONSTUB_AutoCompleteResponse) {
+      return TRITONSERVER_ErrorNew(
+          TRITONSERVER_ERROR_INTERNAL,
+          (std::string(
+               "Received unexpected resposne from Python backend stub: ") +
+           name_)
+              .c_str());
+    }
+
+    auto auto_complete_response =
+        std::move((shm_pool_->Load<AutoCompleteResponseShm>(
+                      auto_complete_response_message->Args())))
+            .data_;
+
+    if (auto_complete_response->response_has_error) {
+      if (auto_complete_response->response_is_error_set) {
+        std::unique_ptr<PbString> error_message =
+            PbString::LoadFromSharedMemory(
+                shm_pool_, auto_complete_response->response_error);
+        throw BackendModelException(TRITONSERVER_ErrorNew(
+            TRITONSERVER_ERROR_INTERNAL, error_message->String().c_str()));
+      } else {
+        throw BackendModelException(TRITONSERVER_ErrorNew(
+            TRITONSERVER_ERROR_INTERNAL,
+            (std::string("Auto-complete failed for ") + Name()).c_str()));
+      }
+    }
+
+    if (auto_complete_response->response_has_model_config) {
+      std::unique_ptr<PbString> auto_complete_config =
+          PbString::LoadFromSharedMemory(
+              shm_pool_, auto_complete_response->response_model_config);
+      std::string auto_complete_config_string = auto_complete_config->String();
+      if (!auto_complete_config_string.empty()) {
+        FixStringToJsonFormat(&auto_complete_config_string);
+        RETURN_IF_ERROR(
+            auto_complete_config_.Parse(auto_complete_config_string));
+        triton::common::TritonJson::WriteBuffer buf;
+        auto_complete_config_.Write(&buf);
+      }
+    }
+
+    initialized_ = true;
+  }
+
+  return nullptr;  // success
+}
+
+void
+ModelState::TerminateStub()
+{
+  Stub()->Destruct(
+      this, &stub_pid_, &initialized_, &stub_message_queue_, &health_mutex_,
+      &parent_message_queue_, &memory_manager_, &ipc_control_, &shm_pool_,
+      nullptr, nullptr, nullptr, false);
+}
+
+void
+ModelState::FixStringToJsonFormat(std::string* str)
+{
+  *str = std::regex_replace(*str, std::regex("\'"), "\"");
+  *str = std::regex_replace(*str, std::regex("False"), "false");
+  *str = std::regex_replace(*str, std::regex("True"), "true");
+}
+
+TRITONSERVER_Error*
+ModelState::AutoCompleteConfig()
+{
+  RETURN_IF_ERROR(CheckModelConfigMismatch());
```

Comment:
With https://github.com/triton-inference-server/python_backend/pull/138/files#r868251707
All this will no longer be here and most of the validation will move to the stub process.
The only checks would be:

1. `set_max_batch_size`: If config.pbtxt specified non-zero value and `set_max_batch_size` tries to set different max_batch_size, throw error.
2.`add_input`:  If config.pbtxt includes input with same name but differs datatype or shape, throw error. 
3. `add_output`: If config.pbtxt includes output with same name but differs datatype or shape, throw error.

Otherwise, it's all good. We don't need any other checks.
CC @Tabrizian

Reply:
Just use the config returned from `GetAutoCompleteConfig()` directly.

Reply:
Sounds good to me!